src/github/command.rs

```diff
@@ -0,0 +1,445 @@
+///          "inquire" | "inquired" | "inquiring" | "inquires" |
+///          "quiz" | "quized" | "quizing" | "quizzes" |
+///          "survey" | "surveyed" | "surveying" | "surveys" ;
+///
+/// team_label ::= "T-lang" | .. ;
+/// team_label_simple ::= "lang" | .. ;
+/// team_ping ::= "@"? "rust-lang/lang" | ..;
+/// team_target ::= team_label | team_label_simple | team_ping ;
+///
+/// line_remainder ::= .+$ ;
+/// ws_separated ::= ... ;
+///
+/// subcommand ::= merge | close | postpone | cancel | review
+///              | concern line_remainder
+///              | resolve line_remainder
+///              | poll [team_target]* line_remainder
+///              ;
+///
+/// invocation ::= "fcp" subcommand
+///              | "pr" subcommand
+///              | "f?" ws_separated
+///              | subcommand
+///              ;
+///
+/// grammar ::= "@rfcbot" ":"? invocation ;
+fn parse_fcp_subcommand<'a>(
+    command: &'a str,
+    subcommand: &'a str,
+    fcp_context: bool
+) -> DashResult<RfcBotCommand<'a>> {
+    Ok(match subcommand {
+        // Parse a FCP merge command:
+        "merge" | "merged" | "merging" | "merges" =>
+            RfcBotCommand::FcpPropose(FcpDisposition::Merge),
+
+        // Parse a FCP close command:
+        "close" | "closed" | "closing" | "closes" =>
+            RfcBotCommand::FcpPropose(FcpDisposition::Close),
+
+        // Parse a FCP postpone command:
+        "postpone" | "postponed" | "postponing" | "postpones" =>
+            RfcBotCommand::FcpPropose(FcpDisposition::Postpone),
+
+        // Parse a FCP cancel command:
+        "cancel" | "canceled" | "canceling" | "cancels" =>
+            RfcBotCommand::FcpCancel,
+
+        // Parse a FCP reviewed command:
+        "reviewed" | "review" | "reviewing" | "reviews" =>
+            RfcBotCommand::Reviewed,
+
+        // Parse a FCP concern command:
+        "concern" | "concerned" | "concerning" | "concerns" => {
+            debug!("Parsed command as NewConcern");
+            RfcBotCommand::NewConcern(parse_command_text(command, subcommand))
+        },
+
+        // Parse a FCP resolve command:
+        "resolve" | "resolved" | "resolving" | "resolves" => {
+            debug!("Parsed command as ResolveConcern");
+            RfcBotCommand::ResolveConcern(parse_command_text(command, subcommand))
+        },
+
+        // Parse an AskQuestion command:
+        "ask" | "asked" | "asking" | "asks" |
+        "poll" | "polled" | "polling" | "polls" |
+        "query" | "queried" | "querying" | "queries" |
+        "inquire" | "inquired" | "inquiring" | "inquires" |
+        "quiz" | "quized" | "quizing" | "quizzes" |
+        "survey" | "surveyed" | "surveying" | "surveys" => {
+            debug!("Parsed command as AskQuestion");
+
+            let mut question = parse_command_text(command, subcommand);
+            let mut teams = BTreeSet::new();
+            while let Some(team_candidate) = question.split_whitespace().next() {
```

Comment:
For some reason this made my brain itch a little -- maybe bc its so uncommon in rust to mutably reassign to the same variable? If it'd be easy I think this would be more nicely expressed as something like:

Reply:
Mmh yeah, I wasn't super happy about these lines too; I'll try to reformulate it.